lib/models/blueprint.js

```diff
@@ -430,12 +443,85 @@ let Blueprint = CoreObject.extend({
     let fileInfos = await process.call(this, intoDir, locals);
 
     // commit changes for each FileInfo (with prompting as needed)
-    await Promise.all(fileInfos.map((fi) => this._commit(fi)));
+    await Promise.all(fileInfos.map((info) => this._commit(info)));
 
     // run afterInstall/afterUninstall userland hooks
     await afterHook.call(this, options);
   },
 
+  /**
+    @private
+    @method shouldConvertToJS
+    @param {Object} options
+    @param {FileInfo} fileInfo
+    @return {Boolean}
+  */
+  shouldConvertToJS(options, fileInfo) {
+    // If this isn't turned on, it doesn't matter what else was passed, we're not touching it.
+    if (!this.shouldTransformTypeScript) {
+      return false;
+    }
+
+    // If the blueprint isn't a TS file to begin with, there's nothing to convert.
+    if (!isTypeScriptFile(fileInfo.outputPath)) {
+      // If the user wants TypeScript output but there is no TypeScript blueprint available, we want
+      // to warn them that they're not going to get what they're expecting while still at least giving
+      // them the JS output. We check for this *after* checking `shouldTranformTypeScript` because
+      // it's possible for people to set `{typescript: true}` in their `.ember-cli` file, which would
+      // then erroneously trigger this message every time they generate a JS blueprint even though
+      // they didn't pass the flag.
+      if (options.typescript === true) {
+        this.ui.writeLine(
+          chalk.yellow(
+            "You passed the '--typescript' flag but there is no TypeScript blueprint available. " +
+              'A JavaScript blueprint will be generated instead.'
+          )
+        );
+      }
+
+      return false;
+    }
+
+    // Indicates when the user explicitly passed either `--typescript` or `--no-typescript` as opposed
+    // to not passing a flag at all and allowing for default behavior
+    const userExplicitlySelectedTypeScriptStatus = options.typescript !== undefined;
+
+    // Indicates when the user has asked for TypeScript either globally (by setting
+    // `isTypeScriptProject` to true) or locally (by passing the `--typescript` flag when they
+    // invoked the generator). Although ember-cli merges `.ember-cli` and all of the flag values into
+    // one object, we thought the DX would be improved by differentiating between what is intended
+    // to be global vs. local config.
+    const shouldUseTypeScript = userExplicitlySelectedTypeScriptStatus
+      ? options.typescript
+      : options.isTypeScriptProject;
+
+    // if the user wants TS output and we have a TS file available, we do *not* want to downlevel to JS
+    if (shouldUseTypeScript && isTypeScriptFile(fileInfo.inputPath)) {
```

Comment:
Aren't we already inside an `isTypeScriptFile` guard from 466 ? (maybe I misread the indentation levels in the diff)

Reply:
Yep, good catch. Removed the guard here.